src/Orc.DataAccess/Database/Gateways/SqlDbSourceGatewayBase.cs

```diff
@@ -91,34 +100,46 @@ public override DbDataReader GetRecords(DataSourceParameters queryParameters, in
 
         private DbCommand CreateGetTableRecordsCommand(DbConnection connection, DataSourceParameters parameters, int offset, int fetchCount)
         {
+            ArgumentNullException.ThrowIfNull(connection);
```

Comment:
technically, since this is private *and* this library enforces nullable, I don't think we need this check here (but it can stay, I think it's something we should discuss).

Reply:
Design choicse: let's keep it here in case of further changes.